Doc/c-api/dict.rst

```diff
@@ -73,7 +82,7 @@ Dictionary Objects
    .. index:: single: PyUnicode_FromString()
 
    Insert *val* into the dictionary *p* using *key* as a key. *key* should
-   be a :c:expr:`const char*`.  The key object is created using
+   be a :c:expr:`const char*` UTF-8 encoded bytes string.  The key object is created using
```

Comment:
It would be nice to backport such docs changes. Could you create a separate PR for this to make backporting easier? There may be other functions with `const char *` parameters without explicitly specified encoding.

Reply:
Once this PR is merged, I will extract the doc change into a separated PR to backport these doc changes.

Reply:
I created PR https://github.com/python/cpython/pull/108448 for the doc changes.